Lib/email/headerregistry.py

```diff
@@ -218,7 +218,7 @@ def __reduce__(self):
                 self.__class__.__bases__,
                 str(self),
             ),
-            self.__dict__)
+            self.__getstate__())
```

Comment:
This is a big branch, so I'll leave just a comment about the email package change: LGTM.